pkg/domain/infra/runtime_abi.go

```diff
@@ -36,7 +36,10 @@ func NewImageEngine(facts *entities.PodmanConfig) (entities.ImageEngine, error)
 	case entities.TunnelMode:
 		// TODO: look at me!
 		ctx, err := bindings.NewConnectionWithIdentity(context.Background(), facts.URI, facts.Identity, facts.MachineMode)
-		return &tunnel.ImageEngine{ClientCtx: ctx}, err
+		if err != nil {
+			return nil, fmt.Errorf("%s: %s", err.Error(), facts.URI)
+		}
```

Comment:
this breaks error wrapping,
callers that try to check the error with errors.Is() or errorsAs() will no longer match the original error.
For example here: https://github.com/containers/podman/blame/d912e735a30ab89e3b9522338afa5f89a73a85f1/cmd/podman/root.go#L120-L122

You have to change this to
```
fmt.Errorf("%w: %s", err, facts.URI)
```